subprojects/core/src/main/java/org/gradle/api/internal/artifacts/configurations/ConfigurationRolesForMigration.java

```diff
@@ -46,32 +47,11 @@ private ConfigurationRolesForMigration() {
     @Deprecated
     public static final ConfigurationRole LEGACY_TO_RESOLVABLE_DEPENDENCY_SCOPE = difference(ConfigurationRoles.ALL, ConfigurationRoles.RESOLVABLE_DEPENDENCY_SCOPE);
 
-    /**
-     * A legacy configuration that will become a consumable configuration in the next major version.
-     */
-    @SuppressWarnings("deprecation")
-    public static final ConfigurationRole LEGACY_TO_CONSUMABLE = difference(ConfigurationRoles.ALL, ConfigurationRoles.CONSUMABLE);
-
-    /**
-     * A resolvable dependency scope that will become a resolvable configuration in the next major version.
-     */
-    @SuppressWarnings("deprecation")
-    public static final ConfigurationRole RESOLVABLE_DEPENDENCY_SCOPE_TO_RESOLVABLE = difference(ConfigurationRoles.RESOLVABLE_DEPENDENCY_SCOPE, ConfigurationRoles.RESOLVABLE);
-
-    /**
-     * A consumable dependency scope that will become a consumable configuration in the next major version.
-     */
-    @SuppressWarnings("deprecation")
-    public static final ConfigurationRole CONSUMABLE_DEPENDENCY_SCOPE_TO_CONSUMABLE = difference(ConfigurationRoles.CONSUMABLE_DEPENDENCY_SCOPE, ConfigurationRoles.CONSUMABLE);
-
     /**
      * All known migration roles.
      */
     public static final Set<ConfigurationRole> ALL = ImmutableSet.of(
-        LEGACY_TO_RESOLVABLE_DEPENDENCY_SCOPE,
-        LEGACY_TO_CONSUMABLE,
-        RESOLVABLE_DEPENDENCY_SCOPE_TO_RESOLVABLE,
-        CONSUMABLE_DEPENDENCY_SCOPE_TO_CONSUMABLE
+        LEGACY_TO_RESOLVABLE_DEPENDENCY_SCOPE
```

Comment:
Has to remain for Detached Configurations.

Reply:
Let's finish these other Configuration-related deprecations and then return to that if there's time.

Reply:
Yeah thats fine we can leave this as-is for now